### PR TITLE
Refresh nova policy with upstream

### PR DESCRIPTION
**Pull request by j2sol**
Many of these had no rules, but upstream now has rules.
Changed order some to match upstream for easier diffing.

### CODE REVIEW

**Comment by j2sol** [comment]
Here is where we differ from upstream:
```
--- policy.json.upstream	2016-05-19 14:27:18.000000000 -0700
+++ policy.json	2016-05-19 14:30:38.000000000 -0700
@@ -1,7 +1,18 @@
 {
-    "context_is_admin":  "role:admin",
+    "context_is_admin":  "role:admin or role:cloud_admin",
     "admin_or_owner":  "is_admin:True or project_id:%(project_id)s",
     "default": "rule:admin_or_owner",
+    "is_projAdmin": "role:project_admin and project_id:%(project_id)s",
+    "is_cloudAdmin": "role:cloud_admin",
+    "member_role": "role:_member_",
+    "projAdmin_or_cloudAdmin": "rule:is_projAdmin or rule:is_cloudAdmin",
+    "is_resource_owner": "project_id:%(project_id)s and user_id:%(user_id)s",
+    "is_owner": "project_id:%(project_id)s",
+    "admin_or_projAdmin" :"role:admin or rule:is_projAdmin",
+    "admin_or_projAdmin_or_cloudAdmin": "role:admin or rule:projAdmin_or_cloudAdmin",
+    "admin_or_projAdmin_or_cloudAdmin_or_owner": "rule:admin_or_projAdmin_or_cloudAdmin or rule:is_resource_owner",
+    "admin_or_projAdmin_or_cloudAdmin_or_member": "rule:admin_or_projAdmin_or_cloudAdmin or rule:member_role",
+    "admin_or_cloudadmin": "role:admin or role:cloud_admin",

     "cells_scheduler_filter:TargetCellFilter": "is_admin:True",

@@ -25,12 +36,12 @@
     "compute:get_diagnostics": "rule:admin_or_owner",
     "compute:get_instance_diagnostics": "rule:admin_or_owner",

-    "compute:start": "rule:admin_or_owner",
-    "compute:stop": "rule:admin_or_owner",
+    "compute:start": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute:stop": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",

     "compute:lock": "rule:admin_or_owner",
     "compute:unlock": "rule:admin_or_owner",
-    "compute:unlock_override": "rule:admin_api",
+    "compute:unlock_override": "rule:admin_or_projAdmin_or_cloudAdmin",

     "compute:get_vnc_console": "rule:admin_or_owner",
     "compute:get_spice_console": "rule:admin_or_owner",
@@ -90,20 +101,20 @@

     "admin_api": "is_admin:True",
     "compute_extension:accounts": "rule:admin_api",
-    "compute_extension:admin_actions": "rule:admin_api",
-    "compute_extension:admin_actions:pause": "rule:admin_or_owner",
-    "compute_extension:admin_actions:unpause": "rule:admin_or_owner",
-    "compute_extension:admin_actions:suspend": "rule:admin_or_owner",
-    "compute_extension:admin_actions:resume": "rule:admin_or_owner",
-    "compute_extension:admin_actions:lock": "rule:admin_or_owner",
-    "compute_extension:admin_actions:unlock": "rule:admin_or_owner",
+    "compute_extension:admin_actions": "rule:admin_or_projAdmin_or_cloudAdmin",
+    "compute_extension:admin_actions:pause": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:unpause": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:suspend": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:resume": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:lock": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:unlock": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
     "compute_extension:admin_actions:resetNetwork": "rule:admin_api",
-    "compute_extension:admin_actions:injectNetworkInfo": "rule:admin_api",
-    "compute_extension:admin_actions:createBackup": "rule:admin_or_owner",
-    "compute_extension:admin_actions:migrateLive": "rule:admin_api",
-    "compute_extension:admin_actions:resetState": "rule:admin_api",
-    "compute_extension:admin_actions:migrate": "rule:admin_api",
-    "compute_extension:aggregates": "rule:admin_api",
+    "compute_extension:admin_actions:injectNetworkInfo": "rule:admin_or_projAdmin_or_cloudAdmin",
+    "compute_extension:admin_actions:createBackup": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:migrateLive": "rule:admin_or_projAdmin_or_cloudAdmin",
+    "compute_extension:admin_actions:resetState": "rule:admin_or_projAdmin_or_cloudAdmin",
+    "compute_extension:admin_actions:migrate": "rule:admin_or_projAdmin_or_cloudAdmin",
+    "compute_extension:aggregates": "role:admin",
     "compute_extension:agents": "rule:admin_api",
     "compute_extension:attach_interfaces": "rule:admin_or_owner",
     "compute_extension:baremetal_nodes": "rule:admin_api",
@@ -122,27 +133,27 @@
     "compute_extension:deferred_delete": "rule:admin_or_owner",
     "compute_extension:disk_config": "rule:admin_or_owner",
     "compute_extension:evacuate": "rule:admin_api",
-    "compute_extension:extended_server_attributes": "rule:admin_api",
+    "compute_extension:extended_server_attributes": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:extended_status": "rule:admin_or_owner",
     "compute_extension:extended_availability_zone": "rule:admin_or_owner",
     "compute_extension:extended_ips": "rule:admin_or_owner",
     "compute_extension:extended_ips_mac": "rule:admin_or_owner",
     "compute_extension:extended_vif_net": "rule:admin_or_owner",
     "compute_extension:extended_volumes": "rule:admin_or_owner",
-    "compute_extension:fixed_ips": "rule:admin_api",
+    "compute_extension:fixed_ips": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:flavor_access": "rule:admin_or_owner",
-    "compute_extension:flavor_access:addTenantAccess": "rule:admin_api",
-    "compute_extension:flavor_access:removeTenantAccess": "rule:admin_api",
+    "compute_extension:flavor_access:addTenantAccess": "rule:admin_or_cloudadmin",
+    "compute_extension:flavor_access:removeTenantAccess": "rule:admin_or_cloudadmin",
     "compute_extension:flavor_disabled": "rule:admin_or_owner",
     "compute_extension:flavor_rxtx": "rule:admin_or_owner",
     "compute_extension:flavor_swap": "rule:admin_or_owner",
     "compute_extension:flavorextradata": "rule:admin_or_owner",
     "compute_extension:flavorextraspecs:index": "rule:admin_or_owner",
     "compute_extension:flavorextraspecs:show": "rule:admin_or_owner",
-    "compute_extension:flavorextraspecs:create": "rule:admin_api",
-    "compute_extension:flavorextraspecs:update": "rule:admin_api",
-    "compute_extension:flavorextraspecs:delete": "rule:admin_api",
-    "compute_extension:flavormanage": "rule:admin_api",
+    "compute_extension:flavorextraspecs:create": "rule:admin_or_cloudadmin",
+    "compute_extension:flavorextraspecs:update": "rule:admin_or_cloudadmin",
+    "compute_extension:flavorextraspecs:delete": "rule:admin_or_cloudadmin",
+    "compute_extension:flavormanage": "rule:admin_or_cloudadmin",
     "compute_extension:floating_ip_dns": "rule:admin_or_owner",
     "compute_extension:floating_ip_pools": "rule:admin_or_owner",
     "compute_extension:floating_ips": "rule:admin_or_owner",
@@ -151,7 +162,7 @@
     "compute_extension:fping:all_tenants": "rule:admin_api",
     "compute_extension:hide_server_addresses": "is_admin:False",
     "compute_extension:hosts": "rule:admin_api",
-    "compute_extension:hypervisors": "rule:admin_api",
+    "compute_extension:hypervisors": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:image_size": "rule:admin_or_owner",
     "compute_extension:instance_actions": "rule:admin_or_owner",
     "compute_extension:instance_actions:events": "rule:admin_api",
@@ -167,8 +178,8 @@
     "compute_extension:networks_associate": "rule:admin_api",
     "compute_extension:os-tenant-networks": "rule:admin_or_owner",
     "compute_extension:quotas:show": "rule:admin_or_owner",
-    "compute_extension:quotas:update": "rule:admin_api",
-    "compute_extension:quotas:delete": "rule:admin_api",
+    "compute_extension:quotas:update": "rule:admin_or_cloudadmin",
+    "compute_extension:quotas:delete": "rule:admin_or_cloudadmin",
     "compute_extension:quota_classes": "rule:admin_or_owner",
     "compute_extension:rescue": "rule:admin_or_owner",
     "compute_extension:security_group_default_rules": "rule:admin_api",
@@ -177,11 +188,11 @@
     "compute_extension:server_groups": "rule:admin_or_owner",
     "compute_extension:server_password": "rule:admin_or_owner",
     "compute_extension:server_usage": "rule:admin_or_owner",
-    "compute_extension:services": "rule:admin_api",
+    "compute_extension:services": "role:admin",
     "compute_extension:shelve": "rule:admin_or_owner",
-    "compute_extension:shelveOffload": "rule:admin_api",
-    "compute_extension:simple_tenant_usage:show": "rule:admin_or_owner",
-    "compute_extension:simple_tenant_usage:list": "rule:admin_api",
+    "compute_extension:shelveOffload": "rule:admin_or_projAdmin_or_cloudAdmin",
+    "compute_extension:simple_tenant_usage:show": "rule:admin_or_projAdmin_or_cloudAdmin or rule:is_owner",
+    "compute_extension:simple_tenant_usage:list": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:unshelve": "rule:admin_or_owner",
     "compute_extension:users": "rule:admin_api",
     "compute_extension:virtual_interfaces": "rule:admin_or_owner",
@@ -192,11 +203,11 @@
     "compute_extension:volume_attachments:create": "rule:admin_or_owner",
     "compute_extension:volume_attachments:update": "rule:admin_or_owner",
     "compute_extension:volume_attachments:delete": "rule:admin_or_owner",
-    "compute_extension:volumetypes": "rule:admin_or_owner",
+    "compute_extension:volumetypes": "rule:admin_or_owner"
     "compute_extension:availability_zone:list": "rule:admin_or_owner",
-    "compute_extension:availability_zone:detail": "rule:admin_api",
+    "compute_extension:availability_zone:detail": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:used_limits_for_admin": "rule:admin_api",
-    "compute_extension:migrations:index": "rule:admin_api",
+    "compute_extension:migrations:index": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:os-assisted-volume-snapshots:create": "rule:admin_api",
     "compute_extension:os-assisted-volume-snapshots:delete": "rule:admin_api",
     "compute_extension:console_auth_tokens": "rule:admin_api",
```

File: roles/nova-common/templates/etc/nova/policy.json [hunk 1]
```diff
@@ -115,95 +116,95 @@
...
-    "compute_extension:availability_zone:list": "",
+    "compute_extension:virtual_interfaces": "rule:admin_or_owner",
+    "compute_extension:virtual_storage_arrays": "rule:admin_or_owner",
+    "compute_extension:volumes": "rule:admin_or_owner",
+    "compute_extension:volume_attachments:index": "rule:admin_or_owner",
+    "compute_extension:volume_attachments:show": "rule:admin_or_owner",
+    "compute_extension:volume_attachments:create": "rule:admin_or_owner",
+    "compute_extension:volume_attachments:update": "rule:admin_or_owner",
+    "compute_extension:volume_attachments:delete": "rule:admin_or_owner",
+    "compute_extension:volumetypes": "rule:admin_or_owner"
```
Comment (lihkin213) [inline roles/nova-common/templates/etc/nova/policy.json]: We are missing a comma here

**Comment by nealbrown** [comment]
👍

**Comment by lihkin213** [comment]
LGTM. Thanks for fixing this!

**Pull request merged**